### ISSUE TITLE
Configuring of shutdownTimeout of AbstractMessageListenerContainer in amqp.rabbit

### ISSUE THREAD

**Issue by vbeskrovnov** [spring-projects/spring-boot#14931]
There is no ability to configure `org.springframework.amqp.rabbit.listener.AbstractMessageListenerContainer#shutdownTimeout`, but it is an important parameter for applications, that use messaging system.
Can we add the property for this field?

**Comment by philwebb** [comment]
@garyrussell Any thoughts on this, does it make sense as a property?

**Comment by philwebb** [comment]
@vbeskrovnov You should be able to use a `AbstractRabbitListenerContainerFactoryConfigurer` bean to customize the value if you need to.

**Comment by garyrussell** [comment]
@philwebb Its's not currently exposed on the container factory, so we'll need a change in Spring AMQP first: https://github.com/spring-projects/spring-amqp/issues/833

Then it can be configured by overriding boot's auto configured factory...

```java
	@Bean
	public SimpleRabbitListenerContainerFactory rabbitListenerContainerFactory(ConnectionFactory connectionFactory,
			SimpleRabbitListenerContainerFactoryConfigurer configurer) {
		SimpleRabbitListenerContainerFactory factory = new SimpleRabbitListenerContainerFactory();
		configurer.configure(factory, connectionFactory);
		factory.setShutDownTimeout(10_000L);
		return factory;
	}
```

We could then consider exposing it as a first class boot property.

**Comment by ayudovin** [comment]
Is it still blocked ?

**Comment by wilkinsona** [comment]
I don't think it is. Thanks for the nudge. The change is in Spring AMQP 2.1.1 so the suggestion from @philwebb to use a configurer is now possible. We now need to consider if we want to add a property or stick with the configured-based approach.
